hazelcast/src/main/java/com/hazelcast/config/MetricsConfig.java

```diff
@@ -0,0 +1,233 @@
+    }
+
+    /**
+     * Sets whether metrics collection should be enabled for the node. If
+     * enabled, Hazelcast Management Center will be able to connect to this
+     * member. It's enabled by default.
+     */
+    @Nonnull
+    public MetricsConfig setEnabled(boolean enabled) {
+        this.enabled = enabled;
+        return this;
+    }
+
+    /**
+     * Returns if metrics collection is enabled.
+     */
+    public boolean isEnabled() {
+        return enabled;
+    }
+
+    /**
+     * Returns whether metrics will be exposed through JMX MBeans.
+     */
+    public boolean isJmxEnabled() {
+        return jmxEnabled;
+    }
+
+    /**
+     * Enables metrics exposure through JMX. It's enabled by default. Metric
+     * values are collected in the {@linkplain #setCollectionIntervalSeconds
+     * metric collection interval} and written to a set of MBeans.
+     */
+    public MetricsConfig setJmxEnabled(boolean jmxEnabled) {
+        this.jmxEnabled = jmxEnabled;
+        return this;
+    }
+
+    /**
+     * Sets the number of seconds the metrics will be retained on the
+     * instance. By default, metrics are retained for 5 seconds (that is for
+     * one collection of metrics values, if default {@linkplain
+     * #setCollectionIntervalSeconds(int) collection interval} is used). More
+     * retention means more heap memory, but allows for longer client hiccups
+     * without losing a value (for example to restart the Management Center).
+     * <p>
+     * This setting applies only to Management Center metrics API. It
+     * doesn't affect how metrics are exposed through JMX.
+     */
+    @Nonnull
+    public MetricsConfig setRetentionSeconds(int retentionSeconds) {
+        Preconditions.checkPositive(intervalSeconds, "retentionSeconds must be positive");
+        this.retentionSeconds = retentionSeconds;
+        return this;
+    }
+
+    /**
+     * Returns the number of seconds the metrics will be retained on the
+     * instance.
+     */
+    public int getRetentionSeconds() {
+        return retentionSeconds;
+    }
+
+    /**
+     * Sets the metrics collection interval in seconds. The same interval is
+     * used for collection for Management Center and for JMX publisher. By default,
+     * metrics are collected every 5 seconds.
+     */
+    @Nonnull
+    public MetricsConfig setCollectionIntervalSeconds(int intervalSeconds) {
+        Preconditions.checkPositive(intervalSeconds, "intervalSeconds must be positive");
+        this.intervalSeconds = intervalSeconds;
+        return this;
+    }
+
+    /**
+     * Returns the metrics collection interval.
+     */
+    public int getCollectionIntervalSeconds() {
+        return this.intervalSeconds;
+    }
+
+    /**
+     * Sets whether statistics for data structures are added to metrics.
+     * It's disabled by default.
+     * <p/>
+     * Note that enabling the data structures metrics also set {@link #minimumLevel}
```

Comment:
A potential typo: `also set` should be `also sets`.

Reply:
Fixed